### DISCUSSION TITLE
Unable to build all packages.

### DISCUSSION THREAD

**Discussion by RameshwarShindeCctech** [iTwin/itwinjs-core#4850] [Q&A]
Hi,

I want to build all packages. Trying with **rush rebuild**. I'm following the instructions from readme.md from the [Repo](https://github.com/iTwin/itwinjs-core). Having timewise latest pull from master also. But facing the issue with two packages to build which are blocking other dependencies.

![image](https://user-images.githubusercontent.com/111970621/209632450-f3a42441-a5bb-44a9-80c8-0e25e722dde8.png)

Am I missing anything? Need help on that ASAP.

**Comment by grigasp** [comment]
These look like two separate issues.

The first one happens due to space characters in your local repository path. Created a PR for the fix: https://github.com/iTwin/itwinjs-core/pull/4860.

I can't reproduce the second one. What node version are you using (`node -v`)?

**Reply by RameshwarShindeCctech** [reply]
Thanks for the reply @grigasp. 

All right. I have removed space characters from my repository. Now had a fresh pull from master, but this time failing with some other packages than the first two I have described.

![image](https://user-images.githubusercontent.com/111970621/210235333-951c858d-4736-408a-b76b-33e9daa9e8c2.png)
![image](https://user-images.githubusercontent.com/111970621/210235381-7d2229ef-44fe-4d21-9862-2cb0c0418f24.png)


I don't have idea why this is happening. Doing no other operations rather than **rush rebuild**.

(The node version I'm using is  **v16.17.1**)

**Reply by grigasp** [reply]
Can you check if this path actually exists?
![image](https://user-images.githubusercontent.com/35135765/210237076-d921273c-9f51-48b8-b46b-24cf938d3819.png)

Executing `rush install` should symlink `@itwin/build-tools` into `core/bentley/node_modules`.

**Reply by RameshwarShindeCctech** [reply]
Yes. @grigasp You are right. That path doesn't exist. So, I re-cloned the repo (maybe the wrong step) and followed rush install n so on. 
And able to build all packages successfully. But **rush cover** is failing its test with 4 packages.

![image](https://user-images.githubusercontent.com/111970621/210312741-7b994fe8-b923-4015-aa3e-774285b7a3b1.png)
![image](https://user-images.githubusercontent.com/111970621/210312792-4b0c27d4-71b6-4c6d-856b-307ba191b87c.png)

Seems like it's also not getting the files. Would you like to suggest anything ahead?

**Reply by grigasp** [reply]
I see two issues there:

- Coverage. I'm not sure about this one, our CI builds pass, so the coverage should be 100%. It would be helpful if you could check the coverage report and see what's not covered.

- Certa. I believe these test packages require you to run `npm run build:ci` rather than just `npm run build`. You can do this for all packages in the repo by running `rush build:ci`.

  @skirby1996, the `README` states that you can do `rush rebuild` and then `rush cover`, which seems to not be true. Should it be updated?

**Reply by aruniverse** [reply]
re the rush cover failure, it expects the webpacked test bundles to exist. Please run `rush webpack:test` prior to running cover. We will update the readme

**Reply by RameshwarShindeCctech** [reply]
@grigasp & @aruniverse Thanks for the help. 

Just to update : Tried with `rush build:ci` and `rush webpack:test` prior to cover, but no luck. Still getting failed. I guess I'm trying ahead if it gets built well in the same way. Will update as so.

**Reply by arnab-git-hub** [reply]
@RameshwarShindeCctech Is there any update with your issue?
